test/zzz-rsyslog.sh

```diff
@@ -224,3 +229,82 @@ os::cmd::expect_success_and_not_text "cat $ops | jq -r .hits.hits[0]._source.lev
 os::cmd::expect_success_and_not_text "cat $ops | jq -r .hits.hits[0]._source.hostname" "^null$"
 ts=$( cat $ops | jq -r '.hits.hits[0]._source."@timestamp"' )
 os::cmd::expect_success "test ${ts} != null"
+
+# Test logrotation (LOG400) - OKD 4.2 and above
+# check rsyslog logs
+rpod=$( get_running_pod rsyslog )
+oc logs $rpod -c rsyslog > $ARTIFACT_DIR/rsyslog.log 2>&1
+oc exec $rpod -c rsyslog -- ls -l /var/log/rsyslog/rsyslog.log > $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+oc exec $rpod -c rsyslog -- cat /var/log/rsyslog/rsyslog.log >> $ARTIFACT_DIR/rsyslog.exec.txt 2>&1
+os::cmd::expect_success_and_text "oc logs $rpod -c rsyslog | grep 'oc exec <pod_name> -- logs'" "^oc exec .pod_name. -- logs$"
+os::cmd::expect_success "oc exec $rpod -c rsyslog -- ls /var/log/rsyslog/rsyslog.log > /dev/null 2>&1"
+logsize=$( oc exec $rpod -c rsyslog -- wc -c /var/log/rsyslog/rsyslog.log | awk '{print $1}' )
+
+# Check if logrotate works as expected
+if [ $logsize -gt 0 ]; then
+    # set max log file count and max log size
+    maxcount=3
+    maxsize=$( expr $logsize / $maxcount )
+    oc set env daemonset/rsyslog LOGGING_FILE_SIZE=$maxsize LOGGING_FILE_AGE=$maxcount
```

Comment:
This will restart the rsyslog pod

Reply:
I think what you should do instead is to 
- shutdown rsyslog
- perform all of the `oc set env`, and configmap changes
- start rsyslog

This will guarantee that rsyslog uses the env and configmap changes when it starts (and avoids test flakes).

We use the same pattern in many places for fluentd - there are shell functions `stop_fluentd` and `start_fluentd` in hack/testing/util.sh which are used in many places for this purpose.  I suggest you do something similar for rsyslog - otherwise, you will need to do the following after using `oc set env ds/rsyslog`:
```bash
rpod=$( get_running_pod rsyslog )
oc set env daemonset/rsyslog LOGGING_FILE_SIZE=$maxsize LOGGING_FILE_AGE=$maxcount
os::cmd::try_until_failure "oc get pods $rpod > /dev/null 2>&1"
os::cmd::try_until_success "oc get pods 2> /dev/null | grep -q 'rsyslog.*Running'"
rpod=$( get_running_pod rsyslog )
```
or do the `oc set env` last:
```bash
edit the configmap - `oc apply`
rpod=$( get_running_pod rsyslog )
oc set env daemonset/rsyslog LOGGING_FILE_SIZE=$maxsize LOGGING_FILE_AGE=$maxcount
os::cmd::try_until_failure "oc get pods $rpod > /dev/null 2>&1"
os::cmd::try_until_success "oc get pods 2> /dev/null | grep -q 'rsyslog.*Running'"
rpod=$( get_running_pod rsyslog )
```